tests/test_npda.py

```diff
@@ -390,3 +392,149 @@ def test_accepts_input_true(self) -> None:
     def test_accepts_input_false(self) -> None:
         """Should return False if NPDA input is rejected."""
         self.assertFalse(self.npda.accepts_input("aaba"))
+
+    def test_show_diagram(self) -> None:
+        """Should construct the diagram for a NPDA"""
+        graph = self.npda.show_diagram()
+        node_names = {node.get_name() for node in graph.nodes()}
+        self.assertTrue(set(self.npda.states).issubset(node_names))
+        self.assertEqual(len(self.npda.states) + 1, len(node_names))
+
+        for state in self.npda.states:
+            node = graph.get_node(state)
+            expected_shape = (
+                "doublecircle" if state in self.npda.final_states else "circle"
+            )
+            self.assertEqual(node.attr["shape"], expected_shape)
+
+        expected_transitions = {
+            ("q0", frozenset({"a, A | ε", "b, B | ε"}), "q1"),
+            (
+                "q0",
+                frozenset(
+                    {
+                        "ε, # | #",
+                    }
+                ),
+                "q2",
+            ),
+            (
+                "q0",
+                frozenset(
+                    {
+                        "a, # | A#",
+                        "a, A | AA",
+                        "a, B | AB",
+                        "b, # | B#",
+                        "b, A | BA",
+                        "b, B | BB",
+                    }
+                ),
+                "q0",
+            ),
+            ("q1", frozenset({"a, A | ε", "b, B | ε"}), "q1"),
+            (
+                "q1",
+                frozenset(
+                    {
+                        "ε, # | #",
+                    }
+                ),
+                "q2",
+            ),
+        }
+        seen_transitions = {
+            (edge[0], frozenset(edge.attr["label"].split("\n")), edge[1])
+            for edge in graph.edges()
+        }
+        self.assertTrue(expected_transitions.issubset(seen_transitions))
```

Comment:
@Vipul-Cariappa not sure why, but this check in the test cases fails locally for me. Since this is an npda, I suspect there is a mistake in the algorithm or test somewhere that's causing non-deterministic failures.

Reply:
@eliotwrobson All the tests pass on my local machine, also on GitHub workflow. Could you please attach the traceback for more context? It would be easier to debug. It will be impossible to debug without it.

Reply:
Here's a screenshot with the error message:
![image](https://github.com/caleb531/automata/assets/1345068/29906e5f-7e24-4917-b594-d655ca1718f5)

The set that's being printed here is `seen_transitions - expected_transitions`, so the transitions that are being produced are not a subset of those that are expected. Not quite sure why this is. Here is console output of me printing the two sets in question:

![image](https://github.com/caleb531/automata/assets/1345068/44e69cac-5119-4825-a324-bc888df4efe2)

Reply:
@Vipul-Cariappa Figured this out and fixed in this commit: https://github.com/caleb531/automata/pull/177/commits/b52775ee5423bd147fa76e568dfca7df7b460619

I think graphviz was adding some extra characters causing the test to fail, but this depends on how this is implemented under the hood. Either way, going to resolve this discussion!